### DISCUSSION TITLE
Quasar Hot Reload launches multiple Electron Apps

### DISCUSSION THREAD

**Discussion by JokerMartini** [quasarframework/quasar#11737] [General - Components / Directives / etc] [answered]
I've recently created a new blank template in Quasar with Electron. When i edit and save a file in the template, it relaunches the Electron app, however it also launches a secondary instance of the same Quasar app. It grows each time, adding another application instance each time. Resulting in a bunch of electron application windows. When I close any of them it closes out everything.

Has anyone else exp this?

**Comment by hawkeye64** [comment]
Sorry, I personally haven't seen this behaviour. Maybe tell us a bit about your environment.

**Reply by JokerMartini** [reply]
What all controls the launching of the applications window for electron apps? It's just electron-main.js right. The file that would actually create the window? I haven't changed any code in that file.

**Reply by JokerMartini** [reply]
I found the issue but it doesn't make any sense to me. I have a helper.js file i wrote and when i import this into electron-main.js 
`import { parseApps, executeAppItem } from "../src/helpers/utils";` it causes the application to launch a duplicate window anytime i change the code in this utils.js file...why would that be. The utils.js file only contains functions, and imports....

```
const fs = require("fs");
const path = require("path");
import os from "os";
const { shell } = require("electron");
const { spawn, spawnSync, exec } = require("child_process");
```

**Reply by hawkeye64** [reply]
Unfortunately, I don't know. I haven't a clue. :(

**Comment by NicolasReibnitz** [comment]
I had exactly the same problem. I ended up making a new project (that didn't have those issues) and transferred all the code piece by piece over to see when it starts breaking again. It never did. So I deleted node_modules and package-lock.json and installed it again. BUT I also saw then that the newly created project used Electron v17, instead of v16. I updated that in the package.json. Now everything works just fine, even in the main project. So, unfortunately, I can't exactly say what fixed it, the Electron update or the new install.
Maybe this helps someone anyway.

**Reply by JokerMartini** [reply]
ill have to give that a try. it's so odd that it keeps happening. im glad im not the only one. i was able to repro it and found what causes it. i was importing another file into my main_electron.js file and for whatever reason when i disabled the import it stopped happening. the file itself is just some general basic helper functions.

**Reply by NicolasReibnitz** [reply]
For me, at least at the end, it did the multiplying Electron instances thing even if I just re-saved the electron-main.js. Even after I removed all imports of other js files.

**Comment by rstoenescu** [comment, marked as answer]
This has been fixed in q/app.

**Reply by JokerMartini** [reply]
im guessing we can then just run an npm update to resolve this issue?

**Reply by NicolasReibnitz** [reply]
Was it the jump to Electron v17?

**Reply by rstoenescu** [reply]
> Was it the jump to Electron v17?

No. It was a bug in q/app. The electron version makes no difference.

**Reply by JokerMartini** [reply]
ok thanks, so for anyone wondering, it appears you should update Quasar to remove the issue

**Reply by rstoenescu** [reply]
"@quasar/app" to be precise which is the App CLI part. "Quasar" is just the UI package. And you should always keep it up to date to benefit from all the new features and fixes.
